### PR TITLE
New feature of Diode 

### PR DESCRIPTION
**Pull request by mayank-kabra2001**
Fixes #

#### Describe the changes you have made in this PR -
Added a new feature of diode 

### Screenshots of the changes (If any) -

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR. 

I am not able to upload the image of diode in circuit elements. Please check it out 

Screenshot from 2020-12-09 15-06-37
Screenshot from 2020-12-09 15-06-41

### CODE REVIEW

File: public/src/modules/Diode.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+/**
+ * @memberof Buffer
+ * Help Tip
+ * @type {string}
+ * @category modules
+ */
+Diode.prototype.tooltipText =
+    "Diode ToolTip : Isolate the input from the output.";
+Diode.prototype.objectType = "Buffer";
```
Comment (gr455) [inline public/src/modules/Diode.js:105]: Change objectType to Diode

File: public/src/modules/Diode.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        inputLength = 1,
+        bitWidth = 1
+    ) {
+
+		super(x, y, scope, dir, bitWidth);
+
+		this.rectangleObject = false;
+        this.setDimensions(15, 15);
+        this.state = 0;
+        this.preState = 0;
```
Comment (gr455) [inline public/src/modules/Diode.js:25]: Is this required?

**Review by gr455 — COMMENTED** [review]
Also fix the 'Simulation stack limit exceeded' error

**Comment by mayank-kabra2001** [comment]
I have resolved all the errors pointed by @gr455 . 
Changes made in 
public/src/circuitelement.js - added a flag and created getter - getDiodeFlag() , setter - setDiodeFlag() .
public/src/engine.js -  used this flag where stack simulation error is called . 
public/src/modules/Diode.js - Added new variables in constructor .

File: public/src/circuitElement.js [hunk 3]
```diff
@@ -22,6 +22,7 @@ import { generateNodeName } from './verilogHelpers';
 export default class CircuitElement {
     constructor(x, y, scope, dir, bitWidth) {
         // Data member initializations
+        const Diodeflag = 0 ;
```
Comment (satu0king) [inline public/src/circuitElement.js:25]: You are making changes to the base class here. That is unacceptable for this feature. Why was this needed?

**Review by satu0king — CHANGES_REQUESTED** [review]
Search for TriState across the entire codebase. Wherever TriState is present, you need to make changes there.

File: public/src/circuitElement.js (hunk 3 above)
Comment (mayank-kabra2001) [reply public/src/circuitElement.js:25]: This is needed to resolve the issue of stack simulation error .

Comment (satu0king) [reply public/src/circuitElement.js:25]: TriState does everything that diode does and more. Refactor this PR similar to how TriState is.
